nir/src/main/scala/scala/scalanative/nir/Types.scala

```diff
@@ -125,6 +125,11 @@ object Type {
       false
   }
 
+  def isPtr(ty: Type): Boolean = ty match {
```

Comment:
Couldn't it be simply `ty.isInstanceOf[Type.Ptr]`?

Reply:
`ty.isInstanceOf[Type.Ptr.type]`?